jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/enricher/WildflyJARHealthCheckEnricherTest.java

```diff
@@ -78,71 +82,170 @@ private void setupExpectations(Map<String, Map<String, Object>> jkubeConfig) {
+        assertThat(probe.getHttpGet().getPort().getIntVal().intValue()).isEqualTo(9990);
+        assertThat(probe.getHttpGet().getPath()).isEqualTo("/health/live");
+        assertThat(probe.getHttpGet().getScheme()).isEqualTo("HTTP");
+
         probe = enricher.getReadinessProbe();
-        assertEquals(9990, probe.getHttpGet().getPort().getIntVal().intValue());
-        assertEquals("/health/ready", probe.getHttpGet().getPath());
-        assertEquals("HTTP", probe.getHttpGet().getScheme());
-        assertNotNull(probe);
+        assertThat(probe.getHttpGet().getPort().getIntVal().intValue()).isEqualTo(9990);
+        assertThat(probe.getHttpGet().getPath()).isEqualTo("/health/ready");
+        assertThat(probe.getHttpGet().getScheme()).isEqualTo("HTTP");
+
+        probe = enricher.getStartupProbe();
+        assertThat(probe.getHttpGet().getPort().getIntVal().intValue()).isEqualTo(9990);
+        assertThat(probe.getHttpGet().getPath()).isEqualTo("/health/started");
+        assertThat(probe.getHttpGet().getScheme()).isEqualTo("HTTP");
     }
 
     @Test
-    public void testWithCustomConfigurationComingFromConf(@Mocked final JavaProject project) {
+    public void testWithCustomConfigurationComingFromConf_withWildflyJarBefore25_0shouldNotAdd_startupProbe() {
+        wildFlyJarDependencyWithVersion("24.1.1.Final");
+
         Map<String, Object> jarConfig = new HashMap<>();
         jarConfig.put("cloud", null);
         final Map<String, Map<String, Object>> config = createFakeConfig("{\"readinessPath\":\"/foo/ready\","
                 + "\"livenessPath\":\"/foo/live\","
                 + "\"port\":\"1234\","
                 + "\"scheme\":\"https\","
                 +"\"livenessInitialDelay\":\"99\","
+                +"\"livenessPeriodSeconds\":\"10\","
+                +"\"readinessInitialDelay\":\"77\","
+                +"\"readinessPeriodSeconds\":\"30\","
+                +"\"startupInitialDelay\":\"10\","
+                +"\"startupPeriodSeconds\":\"10\","
+                +"\"failureThreshold\":\"17\","
+                +"\"successThreshold\":\"27\""
+                + "}");
+        setupExpectations(project, jarConfig, config);
+
+        final WildflyJARHealthCheckEnricher enricher = new WildflyJARHealthCheckEnricher(context);
+        Probe probe = enricher.getLivenessProbe();
+        assertThat(probe).isNotNull();
+        assertThat(probe.getHttpGet().getHost()).isNull();
+        assertThat(probe.getHttpGet().getScheme()).isEqualTo("HTTPS");
+        assertThat(probe.getHttpGet().getPort().getIntVal().intValue()).isEqualTo(1234);
+        assertThat(probe.getHttpGet().getPath()).isEqualTo("/foo/live");
+        assertThat(probe.getInitialDelaySeconds().intValue()).isEqualTo(99);
+        assertThat(probe.getPeriodSeconds().intValue()).isEqualTo(10);
+        assertThat(probe.getSuccessThreshold().intValue()).isEqualTo(27);
+        assertThat(probe.getFailureThreshold().intValue()).isEqualTo(17);
```

Comment:
I see there are a lot of places where we're asserting `Probe` contents. Do you think we should create a new method `assertProbe` that would receive probe to verify and expected values?

Reply:
It would be nice as it was done in  [Quarkus Test](https://github.com/eclipse/jkube/blob/0e8269d1a290a3542a1144a40310e9c39f69d566/jkube-kit/jkube-kit-quarkus/src/test/java/org/eclipse/jkube/quarkus/enricher/QuarkusHealthCheckEnricherTest.java#L285)

Reply:
Should we do something like this?
```java
    private void assertProbe(Probe probe, String scheme, int port, String path, int initialDelaySeconds,
                             int periodSeconds, int successThreshold, int failureThreshold) {
        assertThat(probe)
                .isNotNull()
                .hasFieldOrPropertyWithValue("httpGet.host", null)
                .hasFieldOrPropertyWithValue("httpGet.scheme", scheme)
                .hasFieldOrPropertyWithValue("httpGet.port", new IntOrString(port))
                .hasFieldOrPropertyWithValue("httpGet.path", path)
                .hasFieldOrPropertyWithValue("initialDelaySeconds", initialDelaySeconds)
                .hasFieldOrPropertyWithValue("periodSeconds",  periodSeconds)
                .hasFieldOrPropertyWithValue("successThreshold", successThreshold)
                .hasFieldOrPropertyWithValue("failureThreshold", failureThreshold);
    }
```

Reply:
looks good, but I would move httpGet related assertions to another method `assertHttpGet` and use that here. I see some places where we only check httpGet. This method could be reused there as well.